lib/Sabre/Album/PropFindPlugin.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2022 Robin Appelman <robin@icewind.nl>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Photos\Sabre\Album;
+
+use Sabre\DAV\INode;
+use Sabre\DAV\PropFind;
+use Sabre\DAV\Server;
+use Sabre\DAV\ServerPlugin;
+
+class PropFindPlugin extends ServerPlugin {
+	public const INTERNAL_FILEID_PROPERTYNAME = '{http://owncloud.org/ns}fileid';
+	public const FILE_METADATA_SIZE = '{http://nextcloud.org/ns}file-metadata-size';
+	public const HAS_PREVIEW_PROPERTYNAME = '{http://nextcloud.org/ns}has-preview';
+	public const FAVORITE_PROPERTYNAME = '{http://owncloud.org/ns}favorite';
+	public const DATE_RANGE_PROPERTYNAME = '{http://nextcloud.org/ns}dateRange';
+	public const COVER_PROPERTYNAME = '{http://nextcloud.org/ns}cover';
+	public const NBITEMS_PROPERTYNAME = '{http://nextcloud.org/ns}nbItems';
+	public const LOCATION_PROPERTYNAME = '{http://nextcloud.org/ns}location';
+
+	private Server $server;
+
+	public function initialize(Server $server) {
+		$this->server = $server;
+
+		$this->server->on('propFind', [$this, 'propFind']);
+	}
+
+
+	public function propFind(PropFind $propFind, INode $node) {
+		if ($node instanceof AlbumPhoto) {
+			$propFind->handle('{http://nextcloud.org/ns}file-name', function () use ($node) {
+				return $node->getFile()->getName();
+			});
+
+			$propFind->handle('{http://nextcloud.org/ns}realpath', function () use ($node) {
+				return $node->getFileInfo()->getPath();
+			});
+
+			$propFind->handle(self::INTERNAL_FILEID_PROPERTYNAME, function () use ($node) {
+				return $node->getFile()->getFileId();
+			});
+
+			$propFind->handle(self::FILE_METADATA_SIZE, function () use ($node) {
+				return json_encode($node->getMetadata());
+			});
+
+			$propFind->handle(self::HAS_PREVIEW_PROPERTYNAME, function () use ($node) {
+				return json_encode($node->hasPreview());
+			});
+
+			$propFind->handle(self::FAVORITE_PROPERTYNAME, function () use ($node) {
+				return $node->isFavorite() ? 1 : 0;
+			});
+		}
+
+		if ($node instanceof AlbumRoot) {
+			$propFind->handle(self::COVER_PROPERTYNAME, function () use ($node) {
+				return $node->getCover();
+			});
```

Comment:
**suggestion (non-blocking):** Since the minimum supported PHP version is 7.4 then we should use arrow functions https://www.php.net/manual/en/functions.arrow.php to shorten these handlers and get rid of the use clause
```suggestion
			$propFind->handle(self::COVER_PROPERTYNAME, fn () => $node->getCover());
```

Reply:
Muc cleaner indeed